addons/stock_account/tests/test_stockvaluationlayer.py

```diff
@@ -617,6 +617,49 @@ def test_return_delivery_rounding(self):
         self.assertAlmostEqual(self.product1.value_svl, 25.33)
         self.assertEqual(self.product1.quantity_svl, 2)
 
+    def test_change_quantity_from_other_company(self):
+        """
+        checks that a move on company A from a user logged in company B creates a svl with correct values (the ones from company A)
+        """
+        # Give user access of company A + B, with default A
+        company_A = self.env.company
+        company_B = self.env['res.company'].search([('id', '!=', company_A.id)], limit=1)
+        self.env.user.write({'company_ids': [(6, 0, [company_A.id, company_B.id])], 'company_id': company_A.id})
+        warehouse_B = self.env.user.with_company(company_B)._get_default_warehouse_id()
+        if not warehouse_B:
+            warehouse_B = self.env['stock.warehouse'].sudo().create({'name': 'WH', 'code': 'WH-B', 'company_id': company_B.id})
+            self.assertEqual(self.env.user.with_company(company_B)._get_default_warehouse_id(), warehouse_B)
+        warehouse_A = self.env.user.with_company(company_A)._get_default_warehouse_id()
+        # set product1 property cost method also in comp B
+        self.product1.with_company(company_B).product_tmpl_id.categ_id.property_cost_method = 'average'
+        # make both in moves so that the product has a standard price of 100 in comp A and 10 in comp B
+        self.env.user.company_id = company_A
+        self._make_in_move(self.product1, 1, unit_cost=100)
+        self.env.user.company_id = company_B
+        self._make_in_move(self.product1, 1, unit_cost=10, create_picking=True, loc_dest=warehouse_B.lot_stock_id, pick_type=warehouse_B.in_type_id)
+        # make the cross move
+        cross_move = self.env['stock.move'].create({
+            'name': 'moveC',
+            'product_id': self.product1.id,
+            'location_id': self.supplier_location.id,
+            'location_dest_id': warehouse_A.lot_stock_id.id,
+            'product_uom': self.uom_unit.id,
+            'product_uom_qty': 1,
+            'picking_type_id': warehouse_A.in_type_id.id,
+            'company_id': company_A.id,
+        })
+        picking = self.env['stock.picking'].create({
+            'picking_type_id': cross_move.picking_type_id.id,
+            'location_id': cross_move.location_id.id,
+            'location_dest_id': cross_move.location_dest_id.id,
+            })
+        cross_move.write({'picking_id': picking.id})
+        cross_move._action_confirm()
+        cross_move._action_assign()
+        cross_move.picked = True
+        cross_move._action_done()
```

Comment:
It's also a bit strange to call the confirm/assign/done on the move directly instead of the picking